localstack-core/localstack/services/s3/notifications.py

```diff
@@ -403,7 +403,11 @@ def _verify_target(self, target_arn: str, verification_ctx: BucketVerificationCo
                 QueueName=arn_data["resource"], QueueOwnerAWSAccountId=arn_data["account"]
             )["QueueUrl"]
         except ClientError:
-            LOG.exception("Could not validate the notification destination %s", target_arn)
+            LOG.exception(
+                "Could not validate the notification destination %s",
+                target_arn,
+                exc_info=LOG.isEnabledFor(logging.DEBUG),
+            )
```

Comment:
nit: this stayed `LOG.exception` 

This is very nice, would it be possible to do the following so that we still get why it failed? Sorry it's on me, the current exception logging wasn't that great for this one:
```python
except ClientError as e:
    code = response["Error"]["Code"]
    LOG.error(
        "Could not validate the notification destination %s: %s",
        target_arn,
        code,
        exc_info=LOG.isEnabledFor(logging.DEBUG),
    )
```

So that the user would still get an indication as to why it failed, as this is a Boto error, we can provider at least the name of the exception like "AccessDenied" for example. 

I had started reviewing the rest but saw you put back on draft 👍 will do the rest later

Reply:
Super sorry for the early ping 🙏 There are more incoming changes. Sure 💯  I will fix this one 👍